src/Transports.AspNetCore/WebSockets/SubscriptionList.cs

```diff
@@ -0,0 +1,232 @@
+namespace GraphQL.Server.Transports.AspNetCore.WebSockets;
+
+/// <summary>
+/// A list of subscriptions represented by a dictonary of string keys and <see cref="IDisposable"/> values.
+/// All members are thread-safe.
+/// <br/><br/>
+/// Upon signalling the cancellation token, all future calls to any method except
+/// <see cref="Dispose"/> will throw an <see cref="OperationCanceledException"/>.
+/// <br/><br/>
+/// It is recommended to call <see cref="Dispose"/> immediately after signalling
+/// the cancellation token.
+/// </summary>
+public sealed class SubscriptionList : IDisposable
+{
+    private Dictionary<string, IDisposable>? _subscriptions = new();
+    private readonly object _lock = new();
+
+    /// <summary>
+    /// Disposes all subscriptions in the list.
+    /// </summary>
+    public void Dispose()
+    {
+        List<IDisposable> subscriptionsToDispose;
+        lock (_lock)
+        {
+            if (_subscriptions == null)
+                return;
+            subscriptionsToDispose = _subscriptions.Values.ToList();
+            _subscriptions = null;
+        }
+        foreach (var disposer in subscriptionsToDispose)
+        {
+            try
+            {
+                disposer.Dispose();
+            }
+            catch { }
```

Comment:
I would aggregate failed disposables and eventually throw to give the caller side to know about problem instead of silently swallow all exceptions.

Reply:
Done